### ISSUE TITLE
Error when trying to connect to SQL server

### ISSUE THREAD

**Issue by faahiero** [tareqimbasher/NetPad#215]
After configure the user, pass, select database and check connection, when try to access database structure in Connections Panel, it shows me the following error:

![image](https://github.com/tareqimbasher/NetPad/assets/28544255/b7e1cecd-2ae1-47c1-af60-a1c24966a0f6)
![image](https://github.com/tareqimbasher/NetPad/assets/28544255/05f49b30-7464-42ca-96bd-95cd381ceaa7)

```
One or more errors occurred. (Scaffolding process failed with exit code: 1. dotnet-ef tool output:
Build started...
Build succeeded.
To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
System.PlatformNotSupportedException: Strings.PlatformNotSupported_DataSqlClient
   at Microsoft.Data.SqlClient.SqlConnection..ctor(String connectionString) in /_/artifacts/Project/obj/Release.AnyCPU/Microsoft.Data.SqlClient/netcore/netcoreapp3.1/Microsoft.Data.SqlClient.notsupported.cs:line 386
   at Microsoft.EntityFrameworkCore.SqlServer.Scaffolding.Internal.SqlServerDatabaseModelFactory.Create(String connectionString, DatabaseModelFactoryOptions options)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.ReverseEngineerScaffolder.ScaffoldModel(String connectionString, DatabaseModelFactoryOptions databaseOptions, ModelReverseEngineerOptions modelOptions, ModelCodeGenerationOptions codeOptions)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContextImpl(String provider, String connectionString, String outputDir, String outputDbContextDir, String dbContextClassName, IEnumerable`1 schemaFilters, IEnumerable`1 tableFilters, String modelNamespace, String contextNamespace, Boolean useDataAnnotations, Boolean overwriteFiles, Boolean useDatabaseNames, Boolean suppressOnConfiguring, Boolean noPluarlize)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContext.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Strings.PlatformNotSupported_DataSqlClient)
```
.NET SDK and Runtime installed

![image](https://github.com/tareqimbasher/NetPad/assets/28544255/e8022198-584e-4f70-ac0e-53a1fa88b6b1)

**Comment by faahiero** [comment]
EndeavourOS (Arch Linux based) x64

**Comment by faahiero** [comment]
**EDIT:** After update to 0.7.2, the error changed to:

```One or more errors occurred. (No scaffolded files found in output dir.)``` 
**(Copied from clipboard when clicking the red error messagem in GUI)**



```
stdout: [13:50:37 ERR] (NetPad.Middlewares.ExceptionHandlerMiddleware) An error occurred in request pipeline
System.InvalidOperationException: No scaffolded files found in output dir.
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.GetScaffoldedModelAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 219
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 59
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 67
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkResourcesGenerator.GenerateSourceCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkResourcesGenerator.cs:line 39
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.<>c__DisplayClass13_0.<<GetSourceGeneratedCodeAsync>b__0>d.MoveNext() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 103
--- End of stack trace from previous location ---
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.GetSourceGeneratedCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 129
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkResourcesGenerator.GenerateAssemblyAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkResourcesGenerator.cs:line 57
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.<>c__DisplayClass14_0.<<GetAssemblyAsync>b__0>d.MoveNext() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 159
--- End of stack trace from previous location ---
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.GetAssemblyAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 185
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkDatabaseConnectionMetadataProvider.CreateDbContextAsync(EntityFrameworkDatabaseConnection dbConnection) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkDatabaseConnectionMetadataProvider.cs:line 35
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkDatabaseConnectionMetadataProvider.GetDatabaseStructureAsync(DatabaseConnection databaseConnection) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkDatabaseConnectionMetadataProvider.cs:line 23
   at NetPad.Apps.CQs.GetDatabaseConnectionStructureQuery.Handler.Handle(GetDatabaseConnectionStructureQuery request, CancellationToken cancellationToken) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/CQs/GetDatabaseConnectionStructureQuery.cs:line 17
   at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at NetPad.Services.MediatorRequestPipeline`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.App/Services/MediatorRequestPipeline.cs:line 19
   at NetPad.Plugins.OmniSharp.OmniSharpMediatorPipeline`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next) in /home/tips/Source/TIPS/NetPad/src/Plugins/NetPad.Plugins.OmniSharp/OmniSharpMediatorPipeline.cs:line 48
   at NetPad.Controllers.DataConnectionsController.GetDatabaseStructure(Guid id) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.App/Controllers/DataConnectionsController.cs:line 103
   at lambda_method170(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at NetPad.Middlewares.ExceptionHandlerMiddleware.InvokeAsync(HttpContext httpContext) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.App/Middlewares/ExceptionHandlerMiddleware.cs:line 32
```
**(Copied from terminal execution of AppImage)**

Same environment as before

**Comment by tareqimbasher** [comment]
@faahiero thanks for the update. I wasn't able to replicate this on my system so far, but I was able to replicate on one specific macOS VM. But even there, I can make the error show up only if I try to expand the connection in the tree to get the tables, but if I run a script using that connection it works just fine. Does running a script against the connection work for you?

**Comment by faahiero** [comment]
@tareqimbasher Yes, I tested it now that you mentioned, and the connection works fine when I run scripts against the database, whether pure SQL Queries or LINQ/Lambda Expressions. But the table structure tree continues to show an error. 
Thank you for the tip, now i can query over database!  :)

**Comment by tareqimbasher** [comment]
Thanks for confirming. I'll check out the issue with the tree view. Can you please post `dotnet --info`?

**Comment by RichardNobel** [comment]
Similar issue on my PC, :cry: with *NetPad v0.7.2* running on Windows 11 Pro (23H2).

In the **(New/Edit) Data Connection** dialog window, I can connect to `(localdb)\MSSQLLocalDB` successfully and the local databases are shown in the dropdown list.

But expanding the structure tree doesn't work, nor does running a SQL statement in a Script tab. Both actions result in the same error: *No scaffolded files found in output dir.*

Below I have included the output of `dotnet --info`. Although I'm pretty sure that NetPad is running on .NET 6 ? I believe it showed me a window on first start, which prompted me to install dotnet-ef tools. Now I can't find that window anymore (through the Settings dialog).

# Exception
```c
System.InvalidOperationException: No scaffolded files found in output dir.
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.GetScaffoldedModelAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 219
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 59
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 67
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkResourcesGenerator.GenerateSourceCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkResourcesGenerator.cs:line 39
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.<>c__DisplayClass13_0.<<GetSourceGeneratedCodeAsync>b__0>d.MoveNext() in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 103
--- End of stack trace from previous location ---
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.GetSourceGeneratedCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /home/tips/Source/TIPS/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 129
   at NetPad.ExecutionModel.External.ExternalScriptRunner.GetDataConnectionResourcesAsync(DataConnection dataConnection) in /home/tips/Source/TIPS/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.Setup.cs:line 131
   at NetPad.ExecutionModel.External.ExternalScriptRunner.GetRunDependencies(RunOptions runOptions) in /home/tips/Source/TIPS/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.Setup.cs:line 33
   at NetPad.ExecutionModel.External.ExternalScriptRunner.RunScriptAsync(RunOptions runOptions) in /home/tips/Source/TIPS/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.cs:line 105
```

# Output of `dotnet --info`
```yaml
.NET SDK:
 Version:           8.0.204
 Commit:            c338c7548c
 Workload version:  8.0.200-manifests.7d36c14f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.204\

.NET workloads installed:
 [maui-windows]
   Installation Source: VS 17.9.34728.123
   Manifest Version:    8.0.7/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maui\8.0.7\WorkloadManifest.json
   Install Type:        FileBased

 [maccatalyst]
   Installation Source: VS 17.9.34728.123
   Manifest Version:    17.2.8004/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maccatalyst\17.2.8004\WorkloadManifest.json
   Install Type:        FileBased

 [ios]
   Installation Source: VS 17.9.34728.123
   Manifest Version:    17.2.8004/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.ios\17.2.8004\WorkloadManifest.json
   Install Type:        FileBased

 [android]
   Installation Source: VS 17.9.34728.123
   Manifest Version:    34.0.52/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.android\34.0.52\WorkloadManifest.json
   Install Type:        FileBased


Host:
  Version:      8.0.4
  Architecture: x64
  Commit:       2d7eea2529

.NET SDKs installed:
  8.0.204 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by stephenkirk** [comment]
I'm experiencing a similar issue when trying to run a query against a SQL database. Unlike the original post, the tree view fully loads in my case. Here's the error I'm encountering:

```
System.InvalidOperationException: No scaffolded files found in output dir.
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.GetScaffoldedModelAsync() in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 219
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 59
   at NetPad.Apps.Data.EntityFrameworkCore.Scaffolding.EntityFrameworkDatabaseScaffolder.ScaffoldAsync() in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/Scaffolding/EntityFrameworkDatabaseScaffolder.cs:line 67
   at NetPad.Apps.Data.EntityFrameworkCore.EntityFrameworkResourcesGenerator.GenerateSourceCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/EntityFrameworkCore/EntityFrameworkResourcesGenerator.cs:line 39
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.<>c__DisplayClass13_0.<<GetSourceGeneratedCodeAsync>b__0>d.MoveNext() in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 103
--- End of stack trace from previous location ---
   at NetPad.Apps.Data.FileSystemDataConnectionResourcesCache.GetSourceGeneratedCodeAsync(DataConnection dataConnection, DotNetFrameworkVersion targetFrameworkVersion) in /Users/user248920/Downloads/source/NetPad/src/Apps/NetPad.Apps.Common/Data/FileSystemDataConnectionResourcesCache.cs:line 129
   at NetPad.ExecutionModel.External.ExternalScriptRunner.GetDataConnectionResourcesAsync(DataConnection dataConnection) in /Users/user248920/Downloads/source/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.Setup.cs:line 131
   at NetPad.ExecutionModel.External.ExternalScriptRunner.GetRunDependencies(RunOptions runOptions) in /Users/user248920/Downloads/source/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.Setup.cs:line 33
   at NetPad.ExecutionModel.External.ExternalScriptRunner.RunScriptAsync(RunOptions runOptions) in /Users/user248920/Downloads/source/NetPad/src/Core/NetPad.Runtime/ExecutionModel/External/ExternalScriptRunner.cs:line 105
```

Environment details:
* OS: macOS Sonoma 14.2.1
* Installed .NET runtimes:
```
> dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 8.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 8.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

Note: Normal queries without a connection work fine. For example:
```
var foo = 42;
foo.Dump();
```

This outputs `42` as expected.

**Comment by RichardNobel** [comment]
> Do you mind installing .NET 6 Runtime or SDK, restarting NetPad and see if you still get the same error?

I have installed the .NET 6 SDK. And, after restarting NetPad, now I can select either .NET 8 or .NET 6 on a Script tab.
But unfortunately the same error occurs. ☹️ 

![image](https://github.com/tareqimbasher/NetPad/assets/5920427/d7018355-0e81-4f52-8b3f-aaaf61301fca)

> And you can get to that window you were looking for by clicking the computer icon on the bottom right corner of the main window.

🤔  Could the version (**v8.0.6**) of Entity Framework Core .NET Tool be causing the Scaffolding issue?

![image](https://github.com/tareqimbasher/NetPad/assets/5920427/8f05cbe2-be40-48bb-9f85-3c03f9448bc3)

**Comment by RichardNobel** [comment]
> Could the version (v8.0.6) of Entity Framework Core .NET Tool be causing the Scaffolding issue?

To answer my own question: downgrading dotnet-ef tools _(uninstalling v8, then installing v6),_ doesn't appear to fix the Scaffolding. 🫤 

![image](https://github.com/tareqimbasher/NetPad/assets/5920427/a84942f9-d176-46f4-9d83-8a2cf6f56cfe)

**Comment by tareqimbasher** [comment]
@RichardNobel thank you for testing that and reporting back, I'll take a look at this and get a fix out soon!

**Comment by faahiero** [comment]
> Thanks for confirming. I'll check out the issue with the tree view. Can you please post `dotnet --info`?

Sure, here:

```
.NET SDK:
 Version:           8.0.106
 Commit:            ab2d260803
 Workload version:  8.0.100-manifests.8f6aae10

Ambiente de runtime:
 OS Name:     endeavouros
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/8.0.106/

Cargas de trabalho do .NET instaladas:
 Workload version: 8.0.100-manifests.8f6aae10
 [wasm-tools]
   Origem da Instalação: SDK 8.0.100
   Versão do Manifesto:    8.0.6/8.0.100
   Caminho do Manifesto:       /usr/share/dotnet/sdk-manifests/8.0.100/microsoft.net.workload.mono.toolchain.current/8.0.6/WorkloadManifest.json
   Tipo de Instalação:        FileBased

 [wasm-tools-net6]
   Origem da Instalação: SDK 8.0.100
   Versão do Manifesto:    8.0.6/8.0.100
   Caminho do Manifesto:       /usr/share/dotnet/sdk-manifests/8.0.100/microsoft.net.workload.mono.toolchain.net6/8.0.6/WorkloadManifest.json
   Tipo de Instalação:        FileBased


Host:
  Version:      8.0.6
  Architecture: x64
  Commit:       3b8b000a0e

.NET SDKs installed:
  6.0.131 [/usr/share/dotnet/sdk]
  7.0.120 [/usr/share/dotnet/sdk]
  8.0.106 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.31 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.31 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by victorcomposes** [comment]
@faahiero Could you try upgrading dotnet-ef?

```bash
dotnet tool update --global dotnet-ef --version 8.0.*
```

**Comment by faahiero** [comment]
> @faahiero Could you try upgrading dotnet-ef tool?
> 
> ```shell
> dotnet tool update --global dotnet-ef --version 8.0.*
> ```
> 
> or to the latest 6.0 version
> 
> ```shell
> dotnet tool update --global dotnet-ef --version 6.0.*
> ```


![image](https://github.com/user-attachments/assets/10132fb9-a46a-465c-91f6-5b6fa3ef99b2)

![image](https://github.com/user-attachments/assets/4fbcb017-111f-434e-a848-887d73c2e8b5)

No, still the same! But thanks anyway!

**Comment by tareqimbasher** [comment]
@faahiero @RichardNobel @stephenkirk can you try the [new version](https://github.com/tareqimbasher/NetPad/releases/tag/v0.8.0) and let me know if you're still seeing this problem. If the same issue persists, we should now see more error details.

**Comment by faahiero** [comment]
@tareqimbasher Nice man! I'll try!

**Comment by faahiero** [comment]
At first glance, both Electron and vNext are super fast to startup!

**Comment by tareqimbasher** [comment]
This issue should be fixed as confirmed in #267. Please feel free to reopen if the problem persists with the new version.
